Quarkus tooling should give clear indication to the user when it's not able to retrieve the project java version

### Describe the bug

I have tested a 'quarkus update' of an existing Quarkus 2.15 project. There seems to be an issue related to quarkus-maven-plugin:
```
quarkus update --stream=3.0

ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:3.3.2:update (default-cli) on project quarkus-hibernate: Execution default-cli of goal io.quarkus:quarkus-maven-plugin:3.3.2:update failed: Cannot parse null string -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal io.quarkus:quarkus-maven-plugin:3.3.2:update (default-cli) on project quarkus-hibernate: Execution default-cli of goal io.quarkus:quarkus-maven-plugin:3.3.2:update failed: Cannot parse null string
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:972)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:293)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:196)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)
    at org.apache.maven.wrapper.BootstrapMainStarter.start (BootstrapMainStarter.java:52)
    at org.apache.maven.wrapper.WrapperExecutor.execute (WrapperExecutor.java:161)
    at org.apache.maven.wrapper.MavenWrapperMain.main (MavenWrapperMain.java:73)
Caused by: org.apache.maven.plugin.PluginExecutionException: Execution default-cli of goal io.quarkus:quarkus-maven-plugin:3.3.2:update failed: Cannot parse null string
```
Current quarkus CLI version:
```
quarkus --version
3.3.2
```

[demo.zip](https://github.com/quarkusio/quarkus/files/12565487/demo.zip)


### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

Attaching a demo.zip project.
Unzip it and run the command:
`quarkus update --stream=3.0`

### Output of `uname -a` or `ver`

Linux fedora 6.0.12-100.fc35.x86_64 #1 SMP PREEMPT_DYNAMIC Thu Dec 8 16:53:55 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux

### Output of `java -version`

openjdk version "17.0.2" 2022-01-18

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.15.0.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

mvnw

### Additional information

_No response_

--- a/independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/commands/handlers/UpdateProjectCommandHandler.java
+++ b/independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/commands/handlers/UpdateProjectCommandHandler.java
@@ -21,6 +21,7 @@ import io.quarkus.devtools.commands.data.QuarkusCommandInvocation;
 import io.quarkus.devtools.commands.data.QuarkusCommandOutcome;
 import io.quarkus.devtools.messagewriter.MessageWriter;
 import io.quarkus.devtools.project.BuildTool;
+import io.quarkus.devtools.project.JavaVersion;
 import io.quarkus.devtools.project.QuarkusProject;
 import io.quarkus.devtools.project.QuarkusProjectHelper;
 import io.quarkus.devtools.project.state.ProjectState;
@@ -44,7 +45,16 @@ public class UpdateProjectCommandHandler implements QuarkusCommandHandler {
 
     @Override
     public QuarkusCommandOutcome execute(QuarkusCommandInvocation invocation) throws QuarkusCommandException {
-        invocation.log().info("Detected project Java version: %s", invocation.getQuarkusProject().getJavaVersion());
+        final JavaVersion projectJavaVersion = invocation.getQuarkusProject().getJavaVersion();
+        invocation.log().info("Detected project Java version: %s", projectJavaVersion);
+        if (projectJavaVersion.isEmpty()) {
+            String instruction = invocation.getQuarkusProject().getBuildTool().isAnyGradle() ? "java>targetCompatibility"
+                    : "maven.compiler.release property";
+            invocation.log().error(String.format("Project Java version not detected, set %s to fix the error.", instruction));
+            return QuarkusCommandOutcome.failure();
+        } else {
+            invocation.log().info("Detected project Java version: %s", projectJavaVersion);
+        }
         final ApplicationModel appModel = invocation.getValue(UpdateProject.APP_MODEL);
         final ExtensionCatalog targetCatalog = invocation.getValue(UpdateProject.TARGET_CATALOG);
         final String targetPlatformVersion = invocation.getValue(UpdateProject.TARGET_PLATFORM_VERSION);
@@ -80,7 +90,8 @@ public class UpdateProjectCommandHandler implements QuarkusCommandHandler {
                 final OptionalInt minJavaVersion = extensionsUpdateInfo.getMinJavaVersion();
                 final Optional<Integer> updateJavaVersion;
                 if (minJavaVersion.isPresent()
-                        && minJavaVersion.getAsInt() > invocation.getQuarkusProject().getJavaVersion().getAsInt()) {
+                        && projectJavaVersion.isPresent()
+                        && minJavaVersion.getAsInt() > projectJavaVersion.getAsInt()) {
                     updateJavaVersion = Optional.of(minJavaVersion.getAsInt());
                 } else {
                     updateJavaVersion = Optional.empty();
@@ -184,7 +195,7 @@ public class UpdateProjectCommandHandler implements QuarkusCommandHandler {
             return;
         }
 
-        if (extensionsUpdateInfo.getMinJavaVersion().isPresent()) {
+        if (extensionsUpdateInfo.getMinJavaVersion().isPresent() && project.getJavaVersion().isPresent()) {
             final Integer extensionsMinJavaVersion = extensionsUpdateInfo.getMinJavaVersion().getAsInt();
             if (extensionsMinJavaVersion > project.getJavaVersion().getAsInt()) {
                 log.warn("We detected that some of the updated extensions require an update of the Java version to: %s",

--- a/independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/BuildTool.java
+++ b/independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/BuildTool.java
@@ -47,6 +47,10 @@ public enum BuildTool {
         this.buildFiles = buildFiles;
     }
 
+    public boolean isAnyGradle() {
+        return GRADLE.equals(this) || GRADLE_KOTLIN_DSL.equals(this);
+    }
+
     /**
      * @return {@code \n}-separated lines to add to a {@code .gitignore} file
      */

--- a/independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/JavaVersion.java
+++ b/independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/JavaVersion.java
@@ -25,11 +25,18 @@ public final class JavaVersion {
         return version == null;
     }
 
+    public boolean isPresent() {
+        return version != null;
+    }
+
     public String getVersion() {
         return version;
     }
 
     public int getAsInt() {
+        if (version == null) {
+            throw new IllegalStateException("Version is not available and can't be parsed as an integer.");
+        }
         return Integer.parseInt(version);
     }
 
@@ -50,7 +57,7 @@ public final class JavaVersion {
 
     @Override
     public String toString() {
-        return version;
+        return isEmpty() ? "NA" : version;
     }
 
     // ordering is important here, so let's keep them ordered

